### DISCUSSION TITLE
Introducing Freenet's Ghost Keys

### DISCUSSION THREAD

**Discussion by quasin** [freenet/freenet-core#1189] [General]
Where are the crypto payments donations for keys? The cards do not pass.

**Comment by k4r4b3y** [comment]
Yeah, I was surprised to see "Stripe" as a payment method.  I think Monero (XMR) payment processors fits better with Freenet's vision.  You can use BTCPayServer with its MONERO PLUGIN to accept cheap, fast, and untraceable money payments per the Ghost Key application.  For more info on various Monero payment processors, check out

- https://www.getmonero.org/resources/tools/#payment-gateways
- https://guides.monero.com/docs/tutorials/accept-monero-as-merchant/#for-complex-web-stores-btcpay-server

**Comment by sanity** [comment]
The reason for picking Stripe was to make it accessible to as many people as possible, but if there is demand we can support other payment methods / currencies.

**Reply by k4r4b3y** [reply]
@sanity I would like to get a ghost key using Monero (XMR).  Stripe, or international credit card payments are either unavailable to me, or unnecessarily costly, or (again, unnecessarily) privacy invasive.  I don't want to be part of a legacy payment processor's opaque and incomprehensible data collection (and probably data sharing) practices.

**Comment by adapt-L** [comment]
I don't think you need to do anything complicated like a payment server. Just post a receive address for donations and then ask for users to submit an "OutProof" with the blinded pubkey in the "message" section. I can write a guide and some code to put this together if you want. Just make a note of the TXID so you don't allow them to produce multiple proofs with the same transaction.

See also:
https://feather-wallet.org/docs/how-to-verify-a-transaction-proof/
https://docs.featherwallet.org/guides/formatted-tx-proofs
https://www.getmonero.org/resources/user-guides/prove-payment.html

**Reply by adapt-L** [reply]
Actually, it might be "SpendProof" not "OutProof". I haven't used monero in a while

**Reply by k4r4b3y** [reply]
That is correct.  Freenet project can simply put out a Monero payment address (using Feather Wallet, or the Monero GUI wallet), and then ask for anonymous users to send their "donations" for Ghost Keys to that address.  And then, ask from the anonymous donors for the Monero payment proof.

**Reply by k4r4b3y** [reply]
@adapt-L your first link for feather wallet is broken btw.  Here is the related documentation about Monero payment proofs using Feather Walet:

https://docs.featherwallet.org/guides/prove-payment
